meson.build

```diff
@@ -0,0 +1,222 @@
+endif
+link_args = cc.get_supported_link_arguments(['-prefer-non-pic'])
+
+check_functions = [
+    'strdup',
+    'strndup'
+]
+
+m_dep = cc.find_library('m', required: false)
+if m_dep.found()
+    deps += m_dep
+endif
+
+iconv_dep = cc.find_library('iconv', required: false)
+if iconv_dep.found()
+    deps += iconv_dep
+endif
+if iconv_dep.found() or cc.has_function('iconv_open')
+    conf.set('CONFIG_ICONV', 1)
+endif
+
+# ASM
+nasm_args = []
+nasm = find_program('nasm', required: get_option('asm'))
+enable_asm = nasm.found()
+
+if enable_asm
+    # NASM version check
+    nasm_r = run_command(nasm, '-v')
+    out = nasm_r.stdout().strip().split()
+    if out[1].to_lower() == 'version'
+        if out[2].version_compare('< 2.10')
+            warning('nasm is too old (found @0@); ASM functions are disabled.'.format(out[2]))
+            enable_asm = false
+        endif
+    else
+        warning('Unexpected nasm version string: @0@; disabling.'.format(nasm_r.stdout()))
+        enable_asm = false
+    endif
+endif
+
+# Architecture check
+if host_machine.cpu_family() == 'x86'
+    bittype = '32'
+    nasm_args += '-DARCH_X86_64=0'
+    intel = true
+elif host_machine.cpu_family() == 'x86_64'
+    bittype = '64'
+    nasm_args += ['-DARCH_X86_64=1', '-DPIC']
+    intel = true
+else
+    intel = false
+endif
+
+if enable_asm and not intel
+    warning('ASM functions are not available on non-intel machines; disabling.')
+    enable_asm = false
+elif (not enable_asm) and intel
+    warning('Install nasm-2.10 or later for a significantly faster libass build.')
+endif
+
+if enable_asm
+    conf.set('CONFIG_ASM', 1)
+
+    if host_machine.system() == 'windows'
+        nasm_args += ['-f', 'win' + bittype]
+        if bittype == '64'
+            nasm_args += ['-DHAVE_ALIGNED_STACK=1']
+        else
+            nasm_args += ['-DHAVE_ALIGNED_STACK=0', '-DPREFIX']
+        endif
+    elif host_machine.system() == 'darwin'
+        nasm_args += ['-f', 'macho' + bittype, '-DPREFIX', '-DHAVE_ALIGNED_STACK=1']
+    else
+        nasm_args += ['-f', 'elf' + bittype, '-DHAVE_ALIGNED_STACK=1']
+    endif
+
+    nasm_args += ['-DHAVE_CPUNOP=0', '-Dprivate_prefix=ass']
+    nasm_args += ['-I', meson.current_source_dir() / 'libass' + '/', '-o', '@OUTPUT@', '@INPUT@']
+
+    nasm_gen = generator(nasm, output: '@BASENAME@.o', arguments: nasm_args)
+endif
+
+# Misc deps
+freetype_dep = dependency('freetype', version: '>= 2.2.1',
+                          default_options: ['default_library=' + get_option('default_library'), # workaround for https://github.com/mesonbuild/meson/issues/8047
+                                            'zlib=builtin',
+                                            'png=enabled'])
+if freetype_dep.found()
+    deps += freetype_dep
+    conf.set('CONFIG_FREETYPE', 1)
+endif
+
+fribidi_visibility = ''
+if host_machine.system() == 'windows' and get_option('default_library') == 'static'
```

Comment:
It's quite possible you might need it for the main build? See https://github.com/fribidi/fribidi/blob/c75a94c84ad1c7d3a3df89b42370933976f4ba59/lib/fribidi-common.h#L44-L65

Reply:
I don't remember why I made it Windows-specific, so maybe pkg-config includes it in the flags that are set, and if you find it otherwise (like on Windows) that might not happen? This just makes sure that it doesn't come up as an issue, and maybe it should not be limited to Windows.

Reply:
Surely this is not an issue for main builds because _we don’t build FriBidi (or any other dependency) ourselves in the main build._

Reply:
Ahh, if it's only relevant for building fribidi, then we indeed don't need to care for main builds.
 I thought/feared it may toggle something in public fribidi headers.

Reply:
Oh, hang on. Is it that the build actually fails on Windows if linking against an already-built static FriBidi without the defining `FRIBIDI_LIB_STATIC`? Surely every single dependency should be affected by this, not just FriBidi?

Reply:
@astiob yes, and I believe the reason this isn't an issue for most dependencies is they default to static linking and require a flag to build dynamically.

Reply:
See `HB_DLL_EXPORT` for an example

Reply:
Going to mark resolved under the assumption that my logic here is fine. If you decide on something different for autotools let me know and I'll update, but at the bare minimum it's needed for Windows static builds.

Reply:
The define is needed if we link to a static *fribidi*. Whether or not *libass* is static doesn't matter iiuc. So not entirely correct, but werks in most instances.
We'll likely do something about this for all builds though, so until then this  can stay as is.

Reply:
This arg can be read from the fribidi .pc file (which apparently meson provides access to):
- When fribidi is built only as static, the `Cflags` entry in the .pc file includes `-DFRIBIDI_LIB_STATIC`, so e.g. `pkg-config --cflags fribidi` returns the appropriate value
- When fribidi is built as both static and dynamic, the nonstandard (but supported by `pkgconf`) `CFLAGS.private` entry includes it, so e.g. `pkg-config --cflags fribidi` is appropriate for dynamic linking (appropriate, as that's the usual default when both are present), while `pkgconf --static --cflags fribidi` is appropriate for static linking as an extension. Similarly, `pkgconf --static --cflags libass` will also include `-DFRIBIDI_LIB_STATIC`.

Perhaps meson should add support for this? But also, when building fribidi internally with libass, isn't fribidi supposed to end up built static-only, and thus have `-DFRIBIDI_LIB_STATIC` in its exported cflags? Shouldn't this Just Work, in that case? What's going wrong here?

Reply:
As noted in the second last message in this subthread, this `get_option('default_library') == 'static'` isn't correct to begin with, as the important part is how fribidi was built. Originally, FriBidi did also not mutate its pc file based on build mode, always placing the flag in `Cflags.private`; this only changed recently.
Regardless of FriBidi changes, we did by now adopt the [alluded workaround on our end](https://github.com/libass/libass/commit/c5ee66dfb4b10c3c499aa804c6d1abb195c696e1), so this should just be removed and will always work regardless of the nature of the used FriBidi library, with what it was or wasn't build together with and pkg-config implementation.

Reply:
Was never fully correct, now never needed and thus should be dropped.